app/src/test/java/de/westnordost/streetcomplete/quests/crossing/AddCrossingTest.kt

```diff
@@ -0,0 +1,186 @@
+package de.westnordost.streetcomplete.quests.crossing
+
+import de.westnordost.streetcomplete.quests.TestMapDataWithGeometry
+import de.westnordost.streetcomplete.testutils.node
+import de.westnordost.streetcomplete.testutils.p
+import de.westnordost.streetcomplete.testutils.way
+import org.junit.Assert.*
+import org.junit.Test
+
+class AddCrossingTest {
+    private val questType = AddCrossing()
+
+    @Test fun `free-floating nodes do not count`() {
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1)
+        ))
+        assertEquals(0, questType.getApplicableElements(mapData).toList().size)
+    }
+
+    /*
+      ══╪══
+    */
+    @Test fun `simple crossing counts`() {
+        val shared = node(2, p(0.0, 0.0))
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1, p(0.0, -1.0)),
+            shared,
+            node(3, p(0.0, +1.0)),
+            node(4, p(-1.0, 0.0)),
+            node(5, p(+1.0, 0.0)),
+            way(1, nodes = listOf(1, 2, 3), tags = mapOf("highway" to "unclassified")),
+            way(2, nodes = listOf(4, 2, 5), tags = mapOf("highway" to "footway")),
+        ))
+        assertEquals(shared, questType.getApplicableElements(mapData).toList().single())
+    }
```

Comment:
A 'no' answer for this quest is going to give a weird result: instead of the footway ending at the edge of the road, where pedestrians actually have to stop, it'll continue into the middle of the road. If this result is acceptable, we could do it for #3106 too, but it's not really the right resulting map.

Reply:
Not sure what you mean.

![image](https://user-images.githubusercontent.com/4661658/128019438-ae2745fd-6c79-4e69-87c9-3f697084d900.png)

Reply:
Here's a real [intersection ](https://www.openstreetmap.org/edit#map=20/45.51423/-122.62283)with a footway drawn across the street, but no crossings marked:

![Footway=crossing across the street](https://user-images.githubusercontent.com/83420/128286022-6aef68b0-e45d-4390-9c35-6e33e63ccb2a.png)

Here's the footway after the user gives this quest a "no":

![`crossing=no` node, but still a footway across the street](https://user-images.githubusercontent.com/83420/128286177-7eb1ce07-cdb6-4f57-878c-970149d1a619.png)

And here are the ways drawn as I believe is correct:

![No more way across the street](https://user-images.githubusercontent.com/83420/128286262-b1529554-7b46-419c-8820-986a32687bac.png)

Reply:
Hmm, I understand, you are right.

Reply:
I guess the only solution would be:
1. exclude `footway=crossing`. So basically, the quest is only asked for footways with neither `footway=sidewalk` nor `footway=crossing` tagged
2. but what could be done: For the quest that asks for the type of crossing, also ask for nodes at the intersection between `footway=crossing` and a road, i.e. implicitly assume that footways tagged with crossing are indeed crossings. In the case depicted above - where the whole crossing way was added in error, the user can leave a note, explaining that there is indeed no crossing

Reply:
Yeah, that's similar to what I was doing with #3111: if a footway is drawn to cross a car-way with `sidewalk=separate`, and the surveyor says that it's not actually a crossing, prompt them to leave a note. Even if it doesn't have a `footway=*` key, a 'not a crossing' answer probably means that the ways need to be rearranged.

This stuff is easier on shared-way sidewalks, given StreetComplete's goal to avoid having a quest touch more than one node: separate-way sidewalks just have lots of nodes involved in any road feature.

Reply:
> And here are the ways drawn as I believe is correct:

In such case this quest still resulted in improvement - there will be still a spurious way, but it is not worse than before.

Maybe in such case SC should ask user to produce a note with a photo?